pkg/registries/google/google.go

```diff
@@ -32,50 +34,80 @@ func (g *googleRegistry) Match(image *storage.ImageName) bool {
 }
 
 // Creator provides the type and registries.Creator to add to the registries Registry.
-func Creator() (string, func(integration *storage.ImageIntegration) (types.Registry, error)) {
-	return "google", func(integration *storage.ImageIntegration) (types.Registry, error) {
-		return NewRegistry(integration, false)
-	}
+func Creator() (string, types.Creator) {
+	return "google",
+		func(integration *storage.ImageIntegration, options ...types.CreatorOption) (types.Registry, error) {
+			cfg := types.ApplyCreatorOptions(options...)
+			return NewRegistry(integration, false, cfg.GetGCPTokenManager())
```

Comment:
so this is for all Google services? not just artifact registry?

Reply:
We want to support workload identities for all Google, Azure and AWS services. This PR deals with support for Artifact registry + Google container registry.